File: Practice/AllDomains/Tutorials/30DaysOfCode/Day6LetsReview.cs
using System;
using System.Collections.Generic;
using System.IO;
class Solution 
{
	/*	author:@shivkrthakur  */
    static void Main(String[] args) 
	{
        /* Enter your code here. Read input from STDIN. Print output to STDOUT. Your class should be named Solution */
        int x = Convert.ToInt32(Console.ReadLine().Trim());
        for(int i = 0; i < x; i++)
        {
            string strInput = Console.ReadLine().Trim();
            string evenStrOutput = string.Empty;
            string oddStrOutput = string.Empty;
            
            for(int k = 0; k < strInput.Length; k++)
            {
                if(k % 2 == 0)
                {
                    evenStrOutput += strInput[k];
                }
                else
                {
                    oddStrOutput += strInput[k];
                }
            }
            Console.WriteLine("{0} {1}", evenStrOutput, oddStrOutput);
            
        }
    }
}

File: Practice/AllDomains/Tutorials/30DaysOfCode/Day6LetsReview.js
/*	author:@shivkrthakur  */
var index = 0;
var inputArr = [];
function readLine(){
    return inputArr[index++];
}
function processData(input) {
    //Enter your code here
    inputArr = input.split('\n');
    var t = parseInt(readLine());
    for(var i = 0; i < t; i++){
        var ip = readLine();
        var evenArr = [];
        var oddArr = [];
        var flag = true;
        for(var j = 0; j < ip.length; j++){
            if(flag) evenArr.push(ip[j]);
            else oddArr.push(ip[j]);
            flag = !flag;
        }
        console.log(evenArr.join("") + " " + oddArr.join(""));
    }
}

process.stdin.resume();
process.stdin.setEncoding("ascii");
_input = "";
process.stdin.on("data", function (input) {
    _input += input;
});

process.stdin.on("end", function () {
   processData(_input);
});
